From 986207a34f4c666340f185af4757c56751501e70 Mon Sep 17 00:00:00 2001
From: timyao <tim.yao@amlogic.com>
Date: Tue, 16 Sep 2014 16:45:31 -0700
Subject: [PATCH 4883/5965] PD #94872-2. H264 video decoder: Use kfifo API and
 code clean up for HLS resolution switching. Fix sync fense frame generation
 condition.

Change-Id: I51e51dcc5a62675267aed50b7b6d27a245218a0f
---
 drivers/amlogic/amports/vh264.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/amlogic/amports/vh264.c b/drivers/amlogic/amports/vh264.c
index 3854174e545f..6fc49db5a51f 100755
--- a/drivers/amlogic/amports/vh264.c
+++ b/drivers/amlogic/amports/vh264.c
@@ -1793,7 +1793,7 @@ static void stream_switching_do(struct work_struct *work)
     canvas_t csy,csu,cyd;
 #endif
 
-    if ((!p_last_vf) || (vh264_stream_switching_state != SWITCHING_STATE_OFF)) {
+    if ((!p_last_vf) || (vh264_stream_switching_state == SWITCHING_STATE_OFF)) {
         return;
     }
 
-- 
2.19.0

